router/usecase/optimized_routes_test.go

```diff
@@ -674,6 +674,72 @@ func (s *RouterTestSuite) TestGetCustomQuote_GetCustomDirectQuote_Mainnet_UOSMOU
 	s.validateExpectedPoolIDOneRouteOneHopQuote(quote, expectedPoolID)
 }
 
+// TODO
+func (s *RouterTestSuite) TestGetCustomQuote_GetCustomDirectQuotes_Mainnet_UOSMOUSDC() {
+	config := routertesting.DefaultRouterConfig
+	config.MaxPoolsPerRoute = 5
+	config.MaxRoutes = 10
+
+	var (
+		amountIn = osmomath.NewInt(5000000)
+	)
+
+	mainnetState := s.SetupMainnetState()
+
+	// Setup router repository mock
+	routerRepositoryMock := routerrepo.New()
+	routerRepositoryMock.SetTakerFees(mainnetState.TakerFeeMap)
+
+	// Setup pools usecase mock.
+	poolsUsecase := poolsusecase.NewPoolsUsecase(&domain.PoolsConfig{}, "node-uri-placeholder", routerRepositoryMock, domain.UnsetScalingFactorGetterCb)
+	poolsUsecase.StorePools(mainnetState.Pools)
+
+	routerUsecase := routerusecase.NewRouterUsecase(routerRepositoryMock, poolsUsecase, config, emptyCosmWasmPoolsRouterConfig, &log.NoOpLogger{}, cache.New(), cache.New())
+
+	// Test cases
+	testCases := []struct {
+		name   string
+		poolID []uint64
+
+		// usually it's the number of pools given,
+		// unless any of those pools does not have given asset pair.
+		expectedNumOfRoutes int
+
+		// for single-hop it matches poolID slice
+		expectedPoolID []uint64
+
+		err error
+	}{
+		{
+			name:                "OSMO/USDC - happy case",
+			poolID:              []uint64{1369, 1374, 1484},
+			expectedNumOfRoutes: 3,
+			expectedPoolID:      []uint64{1369, 1374, 1484},
+		},
+		// TODO: 
```

Comment:
For example, pool with ID `1` does not have `OSMO/USDC` pair, what we sholud do?

Reply:
If you are translating some previous example I gave literally, I made it up. You can construct similar example test cases by checking out the pools.

Pool 1 should be `routertesting.UOSMO` and `routestertesting.ATOM`

See:
https://app.osmosis.zone/pool/1

or

```
osmosisd q poolmanager pool 1 --node https://rpc.osmosis.zone:443
```

Reply:
Marking as resolved, list of out denom tokens aswered to this question.